x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/GetEngineAction.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public class GetEngineAction extends ActionType<GetEngineAction.Response> {
+
+    public static final GetEngineAction INSTANCE = new GetEngineAction();
+    public static final String NAME = "cluster:admin/engine/get";
+
+    private GetEngineAction() {
+        super(NAME, GetEngineAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+
+        private final String engineId;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.engineId = in.readString();
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        public Request(String engineId) {
+            this.engineId = engineId;
+        }
+
+        public String getEngineId() {
+            return engineId;
+        }
+    }
+
+    // TODO add CreatedAt, UpdatedAt
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        private final String engineId;
+        private final String[] indices;
+        private final String analyticsCollectionName;
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            this.engineId = in.readString();
```

Comment:
Would it make sense that the Response contains an Engine class? It could write the response using Engine.writeTo() and delegate toXContent to it as well